src/kmod/file-write-tracking.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2016-2020 VMware, Inc.
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ */
+#pragma once
+
+#include "hash-table-generic.h"
+#include "priv.h"
+
+struct file_type_state {
+	enum CB_FILE_TYPE fileType;
+	bool		  didReadType;
+	bool		  isSpecialFile;
+	bool		  try_vfs_read;
+};
+struct file_write_path {
+	char buf[PATH_MAX + 1];
+};
+
+struct file_write_key {
+	struct file *file;
+};
+// Add inode and device when CB_EVENT
+// can handle them on generic file events
+struct file_write_entry {
+	struct HashTableNode   link;
+	struct file_write_key  key;
+	u32		       last_tgid;
```

Comment:
Look like tgid is always a PID. Change type to match.
```suggestion
	pid_t		       last_tgid;
```